internal/compiler-interface/src/main/java/xsbti/AnalysisCallback.java

```diff
@@ -260,4 +261,6 @@ void problem(String what,
     boolean isPickleJava();
 
     Optional<T2<Path, Path>> getPickleJarPair();
+    
+    ReadSourceInfos getSourceInfos();
```

Comment:
I want to mainly request for feedback on whether the public API change is done correctly.

Reply:
Did some more research and ehh... I guess create a `AnalysisCallback3` is the way to go, like what is done in https://github.com/sbt/zinc/pull/1226